timesketch/api/v1/resources/event.py

```diff
@@ -1208,3 +1248,156 @@ def post(self, sketch_id):
         )
 
         return HTTP_STATUS_CODE_OK
+
+
+class EventTagResource(resources.ResourceMixin, Resource):
+    """Resource to add / remove a tag to an event."""
+
+    # The maximum number of events to tag in a single request.
+    MAX_EVENTS_TO_TAG = 500
+
+    # The maximum number of tags in a single request.
+    MAX_TAGS_PER_REQUEST = 500
+
+    @login_required
+    def post(self, sketch_id, event_id):
+        """Add a list of tags to events.
+
+        Not yet implemented.
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+            event_id: Integer primary key for an event database model
+
+        Returns:
+            A JSON object with the event ID and the tag ID (instance of
+            flask.wrappers.Response)
+
+        Raises:
+            HTTP_STATUS_CODE_NOT_FOUND: if sketch or event does not exist
+            HTTP_STATUS_CODE_NOT_IMPLEMENTED: if the method is not implemented
+        """
+        abort(
+            501,  # HTTP_STATUS_CODE_NOT_IMPLEMENTED
+            ("Method not yet implemented. Please use EventTaggingResource instead."),
+        )
+
+    @login_required
+    def delete(self, sketch_id):
+        """
+        Remove tags (max 500) from a list of events (max 500).
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+            in request form:
+                events: list of events to remove tags from
```

Comment:
In the scope of minimizing data transfers, why does the developer need to submit the full event instead of just a list of event ids? In the rest of the method only the ` _event.get("_id")` is ever used anyway.

Benefit of the full event:
If the full event is provided, you don't need a separate parameter for searchindex_id/name, since `event._index` contains the name info already. This would also cover the use-case where tags from multiple events across multiple searchindex shall be removed. (see comment below)